### ISSUE TITLE
`no-restricted-imports`: defer errors when using namespace imports

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/eslint#12261]
**What rule do you want to change?**
`no-restricted-imports`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer and more (?)

**How will the change be implemented? (New option, new default behavior, etc.)?**

**Please provide some example code that this change will affect:**

We want to allow a library to be imported as a namespace, but disallow certain members of that namespace from being used.

```json
"no-restricted-imports": ["error", {
  "name": "lodash-es",
  "importNames": ["zip"]
}]
```

```js
import * as _ from 'lodash-es'; // currently errors

_.zip; // no error here
```

**What does the rule currently do for this code?**
Errors

**What will the rule do after it's changed?**
Don't error unless the banned member is actually used

```js
import * as _ from 'lodash-es'; // no error here

_.zip; // error here!
```

This is possible using `no-restricted-properties`:

```json
"no-restricted-properties": ["error", {
  "object": "_",
  "property": "zip"
}]
```

… but ideally the banned usage would be tied to the import, as opposed to any object with the same name.

**Are you willing to submit a pull request to implement this change?**
Yes

Related: https://github.com/eslint/eslint/issues/10778

**Comment by kaicataldo** [comment]
This sounds useful to me! Curious what the rest of the team thinks.
